TrafficCapture/dockerSolution/src/main/docker/migrationConsole/workflows/captureReplay.yaml

```diff
@@ -0,0 +1,196 @@
+                - name: insecure
+                  value: "{{=fromJSON(inputs.parameters['replayer-target-config'])['allow_insecure']}}"
+                - name: auth-header-value
+                  value: "{{=let a=fromJSON(inputs.parameters['replayer-target-config'])['basic_auth']; a==nil ? '' : 'Basic ' + toBase64(a['username'] + ':' + a['password'])}}"
+                - name: speedup-factor
+                  value: "20.0"
+                - name: kafka-traffic-brokers
+                  value: "{{tasks.get-brokers-list.outputs.parameters.bootstrap-servers}}"
+                - name: kafka-traffic-topic
+                  value: "{{tasks.kafka-topic-setup.outputs.parameters.topic-name}}"
+                - name: kafka-traffic-group-id
+                  value: "{{tasks.id-generator.id}}"
+                - name: replicas
+                  value: "2"
+
+          - name: run-migration-console
+            templateRef:
+              name: migration-console-template
+              template: deploy-console
+            dependencies: [ get-brokers-list ]
+            arguments:
+              parameters:
+                - name: name
+                  value: "diagnostic-console-{{inputs.parameters.session-name}}"
+                - name: source-cluster
+                  value: "{{=replace(inputs.parameters['source-config'], fromJSON(inputs.parameters['source-config'])['endpoint'], tasks['id-generator'].outputs.parameters['proxy-endpoint'])}}"
+                - name: target-cluster
+                  value: "{{inputs.parameters.replayer-target-config}}"
+                - name: kafka-info
+                  value: |
+                    {
+                      "broker_endpoints": "{{tasks.get-brokers-list.outputs.parameters.bootstrap-servers}}",
+                      "standard": ""
+                    }
+                - name: command
+                  value: tail -f /dev/null
+
+          - name: get-user-satisfied-confirmation
+            template: get-user-satisfied-confirmation
+            dependencies: [ proxy-service, deploy-replayer, run-migration-console ]
+
+    - name: get-brokers-list
+      steps: [[]]
+      inputs:
+          parameters:
+            - name: created-bootstrap-servers
+            - name: created-kafka-name
+            - name: provided-kafka-bootstrap-servers
+            - name: provided-kafka-k8s-name
+      outputs:
+        parameters:
+          - name: kafka-name
+            valueFrom:
+              expression: "inputs.parameters['provided-kafka-k8s-name'] == '' ? inputs.parameters['created-kafka-name'] : inputs.parameters['provided-kafka-k8s-name']"
+          - name: bootstrap-servers
+            valueFrom:
+              expression: "inputs.parameters['provided-kafka-bootstrap-servers'] == '' ? inputs.parameters['created-bootstrap-servers'] : inputs.parameters['provided-kafka-bootstrap-servers']"
+
+    - name: get-user-satisfied-confirmation
+      suspend: {}
+
+    - name: id-generator
```

Comment:
Can these functions(?) be pulled out into their own files? really easy to get lost in these files as is